qiskit_nature/second_q/mappers/bosonic_logarithmic_mapper.py

```diff
@@ -0,0 +1,235 @@
+# This code is part of a Qiskit project.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Logarithmic Mapper for Bosons."""
+
+from __future__ import annotations
+import operator
+import math
+import logging
+
+from functools import reduce, lru_cache
+
+import numpy as np
+
+from qiskit.quantum_info import SparsePauliOp
+
+from qiskit_nature.second_q.operators import BosonicOp
+from .bosonic_mapper import BosonicMapper
+
+logger = logging.getLogger(__name__)
+
+
+class BosonicLogarithmicMapper(BosonicMapper):
+    """The Logarithmic boson-to-qubit Mapper.
+
+    This mapper generates a logarithmic encoding of the Bosonic operator :math:`b_k^\\dagger, b_k` to
+    qubit operators (linear combinations of pauli strings).
+    In this logarithmic encoding the number of qubits necessary to represent a bosonic mode is
+    determined by the max occupation :math:`n_k^{max}` of the mode (meaning the number of states used
+    in the expansion of the mode, or equivalently the state at which the maximum excitation can take
+    place). The number of qubits is given by:
+    :math:`\\lceil\\log_2(n_k^{max} + 1)\\rceil`.
+
+    .. note::
+        A consequence of the rounding up for determining the number of required qubits is that the
+        actual max occupation is often larger than the one selected by the user. For example, if the
+        user selects a :math:`n_k^{max} = 2`, then the number of required qubits is
+        :math:`\\lceil\\log_2(3)\\rceil = 2`. If we now compute the max occupation for 2 qubits, we
+        get :math:`2^2 - 1 = 3`, which is larger than the user-selected max occupation. The user should
+        expect that the actual max occupation is always larger than or equal to the one selected.
+        If the code changes the max occupation, the code will issue a warning in the logs.
+
+    The mode :math:`|k\\rangle` is then mapped to the occupation number vector
+    :math:`|0_{n_k^{max}}, 0_{n_k^{max} - 1},..., 0_{n_k + 1}, 1_{n_k}, 0_{n_k - 1},..., 0_{0_k}\\rangle`
+
+    This class implements the equation (34) and (35) of Reference [1].
+
+    .. math::
+        b_k^\\dagger = \\sum_{n_k = 0}^{2^{N_q}-2}(\\sqrt{n_k + 1}|n+1\\rangle\\langle n|)
```

Comment:
```suggestion
        b_k^\\dagger = \\sum_{n_k = 0}^{2^{N_q}-2}\\left(\\sqrt{n_k + 1}|n+1\\rangle\\langle n|\\right)
```
This makes the rendering ever so slightly nicer